dashboard/app/controllers/aichat_controller.rb

```diff
@@ -63,6 +63,32 @@ def log_chat_event
     render(status: :ok, json: response_body)
   end
 
+  # params are studentUserId: number, currentLevelId: number, scriptId: number
+  # POST /aichat/log_chat_event
+  def student_chat_history
+    # Request all chat events for a student at a given level/script.
+    begin
+      params.require([:studentUserId, :currentLevelId, :scriptId])
+    rescue ActionController::ParameterMissing
+      return render status: :bad_request, json: {}
+    end
+
+    # ensure that we have permission to view student's chat events, i.e.,
```

Comment:
Just a heads up that we might need to also get the `script_level_id` from the client if we're viewing chat history on a sublevel, because of some quirks around the sublevel ID being different from the parent script level ID. See logic here for how we handle this in getting/creating a channel token in Lab2:
https://github.com/code-dot-org/code-dot-org/blob/22a3b237d76ac20ab8bb68c021b8da2b16defce2/dashboard/app/controllers/projects_controller.rb#L334-L346

Reply:
Oh good call! Will add handling here and then in a follow-up PR with, will include sending `script_level_id` from client.